src/Security/Authentication/OpenIdConnect/src/Events/PushedAuthorizationContext.cs

```diff
@@ -0,0 +1,59 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.IdentityModel.Protocols.OpenIdConnect;
+
+namespace Microsoft.AspNetCore.Authentication.OpenIdConnect;
+
+/// <summary>
+/// A context for <see cref="OpenIdConnectEvents.PushAuthorization(PushedAuthorizationContext)"/>.
+/// </summary>
+public class PushedAuthorizationContext : PropertiesContext<OpenIdConnectOptions>
+{
+    /// <summary>
+    /// Initializes a new instance of <see cref="PushedAuthorizationContext"/>.
+    /// </summary>
+    /// <inheritdoc />
+    public PushedAuthorizationContext(HttpContext context, AuthenticationScheme scheme, OpenIdConnectOptions options, OpenIdConnectMessage message, HttpRequestMessage parRequest, AuthenticationProperties properties)
```

Comment:
```suggestion
    public PushedAuthorizationContext(HttpContext context, AuthenticationScheme scheme, OpenIdConnectOptions options, OpenIdConnectMessage message, AuthenticationProperties properties)
```

If we think the `PushedAuthorizationContext` needs the `HttpRequestMessage` for some scenario, we should add a property for it instead.

Reply:
I've rearranged this, so we don't need a separate parameter at all. It turned out to be quite inconvenient to deal with the request message, because both the handler and the event might want to make changes to the parameters that will be sent.